### PR TITLE
Accordion onOpen onClose + dropdown keepOpen

### PR DESCRIPTION
**Pull request by zovomat**
added onOpen and onClose event handlers for the accordion, plus the possibility to keep a dropdown open when clicking on a specific item

### CODE REVIEW

**Review by giuliano176 — DISMISSED** [review]

**Review by geeky-abhishek — DISMISSED** [review]

**Review by geeky-abhishek — DISMISSED** [review]

**Review by giuliano176 — DISMISSED** [review]

**Review by nubsthead — DISMISSED** [review]

File: src/components/display/Dropdown.jsx [hunk 1]
```diff
@@ -308,10 +324,19 @@ const Dropdown = React.forwardRef(function DropdownFn(
...
 	const clickOutsidePopper = useCallback(
 		(e) => {
-			dropdownRef.current &&
+			if (
+				dropdownRef.current &&
 				e.target !== dropdownRef.current &&
 				!dropdownRef.current.contains(e.target) &&
+				// check if the attribute is in the event path
+				!reduce(
+					e.path,
```
Comment (beawar) [inline src/components/display/Dropdown.jsx]: e.path in undefined in Firefox, it is a non-standard attribute. This makes the keepOpen not working

File: src/components/display/Dropdown.jsx [hunk 2]
```diff
@@ -308,10 +324,19 @@ const Dropdown = React.forwardRef(function DropdownFn(
 
 	const clickOutsidePopper = useCallback(
 		(e) => {
-			dropdownRef.current &&
+			if (
+				dropdownRef.current &&
 				e.target !== dropdownRef.current &&
 				!dropdownRef.current.contains(e.target) &&
+				// check if the attribute is in the event path
+				!reduce(
```
Comment (beawar) [inline src/components/display/Dropdown.jsx]: Maybe `find` or `some` are better since they stop when a "true" is found

**Review by beawar — CHANGES_REQUESTED** [review]
It might be useful adding 2 lines to explain what keepOpen does in the documentation.

Also, is it wanted that the first commit from @geeky-abhishek is overwritten by @zovomat ?

**Comment by geeky-abhishek** [comment]
> It might be useful adding 2 lines to explain what keepOpen does in the documentation.
> 
> Also, is it wanted that the first commit from @geeky-abhishek is overwritten by @zovomat ?



> 

@beawar ya the actual PR was by @zovomat  I only restored the custom component rendering because the new implementation will break the code wherever we are using it by the old approach

**Comment by beawar** [comment]
> 
> @beawar ya the actual PR was by @zovomat I only restored the custom component rendering because the new implementation will break the code wherever we are using it by the old approach

Oh, my bad, I misread the commits. Ok then, you are absolutely right, maybe that change can be integrated in another PR. Thanks

File: src/components/display/Dropdown.jsx (hunk 1 above)
Comment (zovomat) [reply src/components/display/Dropdown.jsx]: Didn't notice that, thanks

File: src/components/display/Dropdown.jsx (hunk 2 above)
Comment (zovomat) [reply src/components/display/Dropdown.jsx]: I'm kind of ashamed of myself right now, fixing

**Review by geeky-abhishek — APPROVED** [review]

**Review by beawar — APPROVED** [review]

**Review by CerDennis — APPROVED** [review]

**Pull request merged**